src/PrpShop/PRP/Render/QCEMap_Preview.cpp

```diff
@@ -0,0 +1,85 @@
+/* This file is part of PlasmaShop.
+ *
+ * PlasmaShop is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PlasmaShop is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PlasmaShop.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "QCEMap_Preview.h"
+
+#include <PRP/Object/plSceneObject.h>
+#include <PRP/Surface/plCubicEnvironmap.h>
+
+#include <QPainter>
+#include <QToolBar>
+#include <QGridLayout>
+
+
+QCEMap_Preview::QCEMap_Preview(plCreatable* pCre, QWidget* parent)
+                 : QCreatable(pCre, kPreviewCubicEnvironmap, parent)
+{
+    plCubicEnvironmap* obj = plCubicEnvironmap::Convert(fCreatable);
+
+    // Render the faces onto a single cubemap atlas texture
+    auto cubemap = buildCubeMapTexture(obj);
+    
+    fRender = new QPlasmaCubicRender(this, cubemap);
+    fRender->setSizePolicy(QSizePolicy::MinimumExpanding, QSizePolicy::MinimumExpanding);
+    fRender->setMinimumSize(512, 512);
+
+    QToolBar *viewToolbar = new QToolBar(tr("View"), this);
+    viewToolbar->setFloatable(false);
+    auto viewOptions = fRender->createViewOptions();
+    viewToolbar->addWidget(viewOptions);
+
+    QGridLayout* layout = new QGridLayout(this);
+    layout->setContentsMargins(0, 0, 0, 0);
+    layout->setVerticalSpacing(0);
+    layout->addWidget(viewToolbar, 0, 0);
+    layout->addWidget(fRender, 1, 0);
+}
+
+const QImage QCEMap_Preview::buildCubeMapTexture(plCubicEnvironmap *obj) const
+{
+    plMipmap *f = obj->getFace(plCubicEnvironmap::kFrontFace);
```

Comment:
Suggest a more descriptive variable name
```suggestion
    plMipmap *frontFace = obj->getFace(plCubicEnvironmap::kFrontFace);
```